pg/scripts/init.sql

```diff
@@ -52,61 +49,62 @@ create table reactions (
+create index on compounds using gin (string_to_array(name, ';'));
+create index on ec using gist (name, ec);
+create index on organisms (kegg);
+create index on organisms using gist (name);
+create index on domains (abbreviation);
+create index on domains using gist (name);
+
+-- create indexes for the enzymes table
+create index on enzymes (gene, organism, domain, reaction_id, compound);
 create index on enzymes using gin (string_to_array(ec_number, ';'));
 
-
-
--- create search indexes along with materialized views (genes disabled for now for performance reasons)
-create index on compounds using gist(name);
-create index on ec using gist(name);
-create index on reactions using gist(name);
-
--- create materialized view enzymes_genes as select gene from enzymes group by gene;
--- create index on enzymes_genes using gist(gene);
-create materialized view enzymes_organisms as select organism from enzymes group by organism;
-create index on enzymes_organisms using gist(organism);
-create materialized view enzymes_domains as select domain from enzymes group by domain;
-create index on enzymes_domains using gist(domain);
-create materialized view enzymes_reaction_ids as select reaction_id from enzymes group by reaction_id;
-create index on enzymes_reaction_ids using gist(reaction_id);
-create materialized view enzymes_ec_numbers as select unnest(string_to_array(ec_number, ';')) as ec_number from enzymes group by ec_number;
-create index on enzymes_ec_numbers using gist(ec_number);
-create materialized view enzymes_compounds as select compound from enzymes group by compound;
-create index on enzymes_compounds using gist(compound);
-
-
-
 -- create view for fuzzy search through multiple tables and views
 create view multi_search as
 select text 'reaction' as type, kegg as id, name as match from reactions
 union all
-select text 'compound' as type, kegg as id, name as match from compounds
+select text 'reaction' as type, kegg as id, kegg as match from reactions
 union all
-select text 'ec' as type, ec as id, name as match from ec
+select text 'compound' as type, kegg as id, name as match from compounds
 union all
-select text 'organism' as type, organism as id, organism as match from enzymes_organisms
+select text 'compound' as type, kegg as id, kegg as match from compounds
 union all
-select text 'domain' as type, domain as id, domain as match from enzymes_domains
+select text 'ec' as type, ec as id, name as match from ec
 union all
-select text 'reaction' as type, reaction_id as id, reaction_id as match from enzymes_reaction_ids
+select text 'ec' as type, ec as id, ec as match from ec
 union all
-select text 'ec' as type, ec_number as id, ec_number as match from enzymes_ec_numbers
+select text 'organism' as type, kegg as id, name as match from organisms
 union all
-select text 'compound' as type, compound as id, compound as match from enzymes_compounds;
+select text 'domain' as type, abbreviation as id, name as match from domains
```

Comment:
We didn't include the `gene` here before creating the gist index for it was taking too long. I wonder if it would make sense to include it as a part of `multi_search` now since it has its own table.

Reply:
Wouldn't that impact the performance negatively?

Reply:
I just tried this now and it was still taking too long. To answer your question, it would impact the performance of the indexing process negatively but the hope is that it would improve the search performance for the users. If we can get this down to a reasonable time I think it is worth trying. If we can fix the duplicates problems for `genes` maybe we can try this again.

Reply:
Sounds good!

Reply:
It seems that there is not a significant difference between the current number of rows and unique genes `40 016 606` vs `40 002 023` (`14 583` less) so I doubt it would have any significant impact on the indexing. As of result we probably won't be pursuing this in the near future.